src/avx2-emu-funcs.hpp

```diff
@@ -217,55 +217,73 @@ void avx2_emu_mask_compressstoreu64(void *base_addr,
     vtype::mask_storeu(leftStore, left, temp);
 }
 
-template <typename T>
+template <typename T, bool masked = true>
 int avx2_double_compressstore32(void *left_addr,
                                 void *right_addr,
                                 typename avx2_vector<T>::opmask_t k,
                                 typename avx2_vector<T>::reg_t reg)
 {
     using vtype = avx2_vector<T>;
-    const __m256i oxff = _mm256_set1_epi32(0xFFFFFFFF);
 
     T *leftStore = (T *)left_addr;
     T *rightStore = (T *)right_addr;
 
     int32_t shortMask = convert_avx2_mask_to_int(k);
     const __m256i &perm = _mm256_loadu_si256(
             (const __m256i *)avx2_compressstore_lut32_perm[shortMask].data());
-    const __m256i &left = _mm256_loadu_si256(
-            (const __m256i *)avx2_compressstore_lut32_left[shortMask].data());
 
     typename vtype::reg_t temp = vtype::permutevar(reg, perm);
 
-    vtype::mask_storeu(leftStore, left, temp);
-    vtype::mask_storeu(rightStore, _mm256_xor_si256(oxff, left), temp);
+    if constexpr (masked) {

```

Comment:
Do we need the masked store? I think we can get away with just the regular store. Or am I wrong?